### ISSUE TITLE
Jsonnet support in Benthos

### ISSUE THREAD

**Issue by MrSplinterRat** [Jeffail/benthos#707]
Hello!

Could you consider adding Jsonnet (https://jsonnet.org/) in 'Processors' ?

1. This is a very common Json-processing language, there are many production-ready libraries for it, it is often used in configuration and CI-CD pipelines
2. There is a mature implementation on Golang from Google, https://github.com/google/go-jsonnet - it only remains to implement it in Benthos

Thanks)

**Comment by Jeffail** [comment]
Hey @MrSplinterRat, totally, that seems like a reasonable fit. We could even add this as a Bloblang function, maybe it would look like this:

```coffee
root = jsonnet("""{
  person1: {
    name: "Alice",
    welcome: "Hello " + self.name + "!",
  },
  person2: self.person1 { name: "Bob" },
}""")
```

Which would allow you to do stuff like:

```coffee
root = this.merge(jsonnet(file("./stuff/doc.libsonnet"))
```

**Comment by MrSplinterRat** [comment]
Using jsonnet from within Bloblang seems like a very nice and sane idea to me, and it will be useful in many cases. But along with this, I think it would be nice if Jsonnet also appeared in the list of 'Processors' as it is - similar to how it is now done for Awk, jq and Sql. Thus, it will be possible to use both from Bloblang-scripts and as an independent Processor. I see this as the best solution. Do you see it the same way?

**Comment by illotum** [comment]
I was under the impression the Processors API is subject to change in V4. Adding it to processors means more code to port.

**Comment by Jeffail** [comment]
That should be fine, it'll be easy enough to migrate the current style over for internal code so I wouldn't say this is blocked on v4.

**Comment by twmb** [comment]
Closing as part of a backlog cleanup. This has been open since 2021 with no further activity. Jsonnet support was never implemented. If this is still needed, feel free to open a new issue or PR.
